### DISCUSSION TITLE
Invalid response from GraphQL API on timeout

### DISCUSSION THREAD

**Discussion by autarch** [community/community#78614] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

When the GraphQL API times out generating a response, it seems to return a body like this:

```
{"message": "We couldn't respond to your request in time. Sorry about that. Please try resubmitting your request and contact us if the problem persists."}
```

There's a couple issues with this. First, AFAICT it isn't documented anywhere.

But more importantly, from reading of [the spec](https://spec.graphql.org/October2021/#sec-Response), this is not a valid GraphQL response. The spec says this:

> If the request raised any errors, the response map must contain an entry with key `errors`. The value of this entry is described in the “Errors” section. If the request completed without raising any errors, this entry must not be present.
>
> If the request included execution, the response map must contain an entry with key `data`. The value of this entry is described in the “Data” section. If the request failed before execution, due to a syntax error, missing information, or validation error, this entry must not be present.

By my reading, this says that the response _must_ contain either a `data` or `errors` key. There's nothing in the spec about a `message` key. While ideally clients would detect this invalid response, they may not. And indeed, the library I was using just tried to parse this response into a struct that contains `data` and `errors`, which was _very_ confusing for a long time!

I think it'd be good to fix this. As a compromise, GitHub could leave the `message` key and add an `errors` key as well.

**Comment by jsoref** [comment]
You aren't alone, the `gh` client doesn't handle it well.

> gh: We couldn't respond to your request in time. Sorry about that. Please try resubmitting your request and contact us if the problem persists. (HTTP 504)


According to:
https://youtrack.jetbrains.com/issue/IJPL-72910/GitHub-Pull-requests-UI-Error-on-PRs-with-lost-of-large-commits-on-large-repos-with-long-history

It's actually a `504 Gateway Timeout`. In order for GitHub to fix this (which they kinda should), their gateway would need to have special handling for `504` to the graphql endpoint and return a different message.


As you raised this over a year ago, it doesn't appear that this forum works for feedback, so I'd encourage everyone to use https://support.github.com/contact/bug-report instead.

- https://github.com/googleapis/release-please/issues/2265

**Comment by autarch** [comment]
I filed a bug report. We'll see if that gets any notice.
